jscomp/test/record_regression.js

```diff
@@ -60,6 +60,110 @@ function setAA(ao) {
         };
 }
 
+var ir0 = {
+  TAG: /* V0 */0,
+  x0: "v0",
+  x3: 3
+};
+
+var ir1 = {
+  TAG: /* V0 */0,
+  x0: "v0",
+  x1: "v1",
+  x3: 3
+};
+
+var ir2 = {
+  TAG: /* V0 */0,
+  x0: "v0",
+  x1: "v1",
+  x2: 2,
+  x3: 3
+};
+
+var ir3 = {
+  TAG: /* V1 */1,
+  y0: "v0",
+  y1: 1
+};
+
+var pm0;
+
+pm0 = ir0.TAG === /* V0 */0 ? [
```

Comment:
Side comment: not relevant to this PR, but not sure why assigning value later after declaring the variable first.

Reply:
```rescript
type bar = V0({x: string}) | V1({y: string})
let v = switch bar0 { ... }
```
It seems that pattern matching of variants with inlined records is generating accordingly. Not relevant to the optional inlined record.